azure/services/scalesetvms/client.go

```diff
@@ -74,79 +59,56 @@ func newVirtualMachineScaleSetVMsClient(subscriptionID string, baseURI string, a
 }
 
 // Get retrieves the Virtual Machine Scale Set Virtual Machine.
-func (ac *azureClient) Get(ctx context.Context, resourceGroupName, vmssName, instanceID string) (compute.VirtualMachineScaleSetVM, error) {
+func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "scalesetvms.azureClient.Get")
 	defer done()
 
-	return ac.scalesetvms.Get(ctx, resourceGroupName, vmssName, instanceID, "")
+	return ac.scalesetvms.Get(ctx, spec.ResourceGroupName(), spec.OwnerResourceName(), spec.ResourceName(), "")
 }
 
-// GetResultIfDone fetches the result of a long-running operation future if it is done.
-func (ac *azureClient) GetResultIfDone(ctx context.Context, future *infrav1.Future) (compute.VirtualMachineScaleSetVM, error) {
-	ctx, _, spanDone := tele.StartSpanWithLogger(ctx, "scalesetvms.azureClient.GetResultIfDone")
-	defer spanDone()
+// CreateOrUpdateAsync is a dummy implementation to fulfill the async.Reconciler interface.
+func (ac *azureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, parameters interface{}) (result interface{}, future azureautorest.FutureAPI, err error) {
```

Comment:
Just for my own understanding: Is the Creating/Updating handled by the MachinePool service and that's why no implementation of CreateOrUpdateAsync is needed?

Reply:
Yep!